cmd/otelcorecol/go.mod

```diff
@@ -81,6 +82,8 @@ require (
 
 replace go.opentelemetry.io/collector => ../../
 
-replace go.opentelemetry.io/collector/semconv => ../../semconv
+replace go.opentelemetry.io/collector/confmap => ../../confmap
```

Comment:
Not sure I get the last part of your comment, but I see two options in dealing with -contrib: the first is doing each module here in one PR and doing the -contrib counterpart following before doing another core module. This would be required only for the "core API", not for the components that we have in core (otlp receiver/exporter, processors, ...).

I think I would prefer small steps so that even a lot of touches to the contrib modules would be manageable changes.

Reply:
The `multimod` (a.k.a version tool https://github.com/open-telemetry/opentelemetry-collector/blob/main/versions.yaml since uses a version.yaml :)) must add all these replace statements.

Reply:
contrib makes use of `crosslink` in a few places, is this something we could use here?

Reply:
That would be great, I heard that crosslink does not resolve dependency and blindly adds all replaces, may be good to look into fixing that, but happy to start with that tool.

Reply:
@jpkrohling are you going to take as an action item to look into `crosslink`?

Reply:
I can certainly create an item for that.

Reply:
https://github.com/open-telemetry/opentelemetry-collector/issues/6230

Reply:
Understood. I'm a bit slower now, but I should still be able to get it done.

Reply:
Crosslink [PR](https://github.com/open-telemetry/opentelemetry-collector/pull/6237) is up.